ibmvmc/ibmvmc.c

```diff
@@ -655,12 +662,13 @@ static int ibmvmc_close(struct inode *inode, struct file *file)
 			return -EIO;
 		}
 
-		/* TODO locking? */
+		spin_lock_irqsave(&hmc->lock, flags);
```

Comment:
Partially because of the TODO, but the intention was to prevent sending close CRQs for the HMC if the HMC was already returned by the reset task. (We also do a state check and CRQ send under the HMC lock in ibmvmc_write)